library/vyos_facts.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The module file for vyos_facts
+"""
+
+from __future__ import absolute_import, division, print_function
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+from ansible.module_utils. \
+     vyos.facts.facts import Facts
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': [u'preview'],
+                    'supported_by': 'network',}
+
+
+DOCUMENTATION = """
+---
+module: vyos_facts
+version_added: 2.9
+short_description: Get facts about vyos devices.
+description:
+  - Collects facts from network devices running the vyos operating
+    system. This module places the facts gathered in the fact tree keyed by the
+    respective resource name.  The facts module will always collect a
+    base set of facts from the device and can enable or disable
+    collection of additional facts.
+author: Nilashish Chakraborty (@nilashishc)
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset. Possible values for this argument include
+        all, min, hardware, config, legacy, and interfaces. Can specify a
```

Comment:
Can you add these as choices:  ..or are there more choices than what you have listed here? If so, is it worth saying 'see vendor documentation for full list of values? If there is something in VyOS documentation that has that list?

Reply:
@samccann So this has been copied over from vyos_facts doc in ansible/ansible. I believe we didn't add `choices` in doc and in module argspec because there can be a permutation and combination of these choices. `gather_subset` is a list essentially which can take multiple choices and those prepended with `!` are ignored.

Reply:
ok thanks!

Reply:
cool thanks!